### PR TITLE
[Yaml] fix parsing inline YAML spanning multiple lines

### LINKED ISSUE #25239: [YAML] Nested array incorrectly parsed
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.3.13

Given

```php
$yaml = <<<YAML
services:
    foo:
        class: Foo
        arguments: ['@bar', '@baz', [
            '@qux',
            '%foo%',
            bar,
            123
        ]]

YAML;

dump(Symfony\Component\Yaml\Yaml::parse($yaml));
```

I get

```
Malformed inline YAML string: ['@bar', '@baz', [ '@qux', '%foo%', bar, 123 at line 8 (near " 123").
```

While this is valid at http://www.yamllint.com/

To make it work in SF i had to change the last lines to `123]]`

cc @xabbuh :-)

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #25239 #25379 #31333
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Yaml/Tests/ParserTest.php [hunk 1]
```diff
@@ -1614,6 +1615,206 @@ public function multiLineDataProvider()
...
+            'single quoted multi-line string' => [
+                "foo\nbar",
+                <<<YAML
+'foo
+
+bar'
+YAML
+                ,
+            ],
+            'double quoted multi-line string' => [
```
Comment (javiereguiluz) [inline src/Symfony/Component/Yaml/Tests/ParserTest.php]: The YAML content of this test doesn't match the description. It should use double quotes, right?

**Review by javiereguiluz — APPROVED** [review]
Excellent work! Thanks Christian!

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Comment by mxr576** [comment]
Any plans for back-porting this to 3.4.x? 
I created a smaller patch from this - without tests - that could be applied on 3.4.x (required by Drupal 8.9.x)

[patch.diff.txt](https://github.com/symfony/symfony/files/4917720/patch.diff.txt)

**Comment by xabbuh** [comment]
We never add new features in patch releases and since we didn't support the inline notation across multiple lines before this was considered to be a new feature.

**Comment by mxr576** [comment]
Thanks for the quick update, so we and others can use the patch above if they get a validation error with 3.4.x.

**Pull request merged**